tests/end_to_end/test_client_json.py

```diff
@@ -91,27 +95,127 @@ def verifyParsedOutputsEqual(
 
-    def testSearchAllReferenceSets(self):
+    def testSearchReadGroupSets(self):
+        for dataset in self._client.searchDatasets():
+            iterator = self._client.searchReadGroupSets(dataset.id)
+            self.verifyParsedOutputsEqual(
+                iterator, "readgroupsets-search",
+                "--datasetId {}".format(dataset.id))
+
+    def testSearchReads(self):
+        start = 0
+        end = 1000
+        for dataset in self._client.searchDatasets():
+            for readGroupSet in self._client.searchReadGroupSets(dataset.id):
+                for readGroup in readGroupSet.readGroups:
+                    reference = self._client.searchReferences(
+                        referenceSetId=readGroup.referenceSetId).next()
+                    referenceId = reference.id
+                    iterator = self._client.searchReads(
+                        [readGroup.id], referenceId=referenceId,
+                        start=start, end=end)
+                    args = "--start {} --end {} --readGroupIds {}\
+                    --referenceId {}".format(
+                        start, end, readGroup.id, referenceId)
+                    self.verifyParsedOutputsEqual(
+                        iterator, "reads-search", args)
+
+    def testSearchReferenceSets(self):
         iterator = self._client.searchReferenceSets()
         self.verifyParsedOutputsEqual(iterator, "referencesets-search")
 
-    def testSearchAllReferences(self):
+    def testSearchReferences(self):
         for referenceSet in self._client.searchReferenceSets():
             iterator = self._client.searchReferences(
                 referenceSetId=referenceSet.id)
             args = "--referenceSetId={}".format(referenceSet.id)
             self.verifyParsedOutputsEqual(iterator, "references-search", args)
 
-    def testGetDataset(self):
+    def testSearchVariantSets(self):
         for dataset in self._client.searchDatasets():
-            self.verifyParsedOutputsEqual(
-                [dataset], "datasets-get", dataset.id)
+            iterator = self._client.searchVariantSets(dataset.id)
+            self.verifyParsedOutputsEqual(iterator, "variantsets-search")
 
-    def testSearchAllReadGroups(self):
-        # TODO: add more rigorous testing here
-        cliOutput = self.captureJsonOutput("reads-search")
-        self.assertGreater(len(cliOutput), 0)
+    def testSearchVariants(self):
+        start = 0
+        end = 1000
+        referenceName = "1"
+        for dataset in self._client.searchDatasets():
+            for variantSet in self._client.searchVariantSets(dataset.id):
+                iterator = self._client.searchVariants(
+                    variantSet.id, start=start, end=end,
+                    referenceName=referenceName, callSetIds=[])
+                args = "--variantSetId {} --start {} --end {} -r {}".format(
+                    variantSet.id, start, end, referenceName)
+                self.verifyParsedOutputsEqual(
```

Comment:
Can we do a ``tests_executed`` (or something similar) counter here? I think it's important to make sure we're not testing entirely empty output here.